Docs: Edit example README and add comments in Basic Panel Plugin (#236)
grafana/grafana-plugin-examples: examples/panel-basic/README.md

# Grafana Panel Plugin

> This example provides a template for how to build a basic panel plugin. The entry point is `/src/components/SimplePanel/SimplePanel.tsx`.

Reviewer: I'm not sure this statement is correct. The real entry of the plugin is the `module.ts` but the panel is defined in `SimplePanel.tsx`.

Maybe we can just find a different wording instead of entry point?

Author: changed this to "The definition for the panel is provided in `/src/components/SimplePanel/SimplePanel.tsx`."

> By default, the plugin uses `TimeSeries` from the `grafana-ui` package to build a graph with the properties of `SimplePanel`. The plugin also allows a tooltip to be shown when user hovers over a visualization.

Reviewer: ```suggestion
The plugin uses `TimeSeries` from the `grafana-ui` package to build a graph with the properties passed to the panel. The plugin also allows a tooltip to be shown when user hovers over a visualization.
```

Author: Done.

> Additionally, the plugin is configured to allow features such as a gradient mode selector and a list display mode on the Grafana sidebar.

Reviewer: ```suggestion
Additionally, the plugin is setup to allow custom options such as gradient mode selector and a list display mode to be configured from the Grafana sidebar.
```

Author: Done.

## What is a Grafana panel plugin?

Panel plugins allow you to add new types of visualizations to your dashboard, such as maps, clocks, pie charts, lists, and more.

Use panel plugins when you want to do things like visualize data returned by data source queries, navigate between dashboards, or control external systems (such as smart home devices).

## Getting started

1. Install dependencies:

   ```bash

>    yarn install

Reviewer 2: not part of your PR but this bit is wrong in all the examples. this should be `npm install`. we are not using yarn in any of the examples anymore.

Reviewer: I created a separate issue for this https://github.com/grafana/grafana-plugin-examples/issues/237

Author: I've fixed it in this PR.

>    yarn install

Reviewer 2: this line must be `npm install`. We should not install dependencies with yarn because we use npm in this repository.

   ```

2. Build the plugin in development mode and run inside Grafana using Docker:

   ```bash
   # Start watching for changes

>    npm dev

Reviewer 2: Except for `npm install` all of these should `npm run [command]` or we'll get an error.

here should be `npm run dev`  and down the line `npm run build` and `npm run e2e`

   docker-compose up
   ```

3. Build plugin in production mode:

   ```bash
   yarn build
   ```

4. Run e2e tests:

   ```bash

>    yarn e2e

Reviewer 2: `npm run e2e`
